### PR TITLE
oauth2: recognise different identities by GrantType

### LINKED ISSUE #21: oauth2: userInfo endpoint should return user info for service accounts
When userInfo is requested with token issued for service account it should return proper information for it. Currently interface is design as follow: 

``` java
public interface IdentityFinder {
  Optional<Identity> findIdentity(String identityId, DateTime instantTime);
}
```

and the problem is that implementations wouldn't be able easily to distinguish between identities of service accounts and the resource owners.

### PR DESCRIPTION
**Pull request by mgenov**
Token now contains and grant type which could be used by the applications to separate Service Accounts from Resource Owners.

Fixes #21

### CODE REVIEW

File: oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryUserRepository.java [hunk 1]
```diff
@@ -27,7 +28,7 @@
   }
 
   @Override
-  public Optional<Identity> findIdentity(String identityId, DateTime instantTime) {
+  public Optional<Identity> findIdentity(String identityId, GrantType grantType, DateTime instantTime) {
```
Comment (tsony-tsonev) [inline oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryUserRepository.java:31]: Here at the implementation in the example app we could make a simple usage of the grantType with the Identity just to show its purpose and make the idea clearer.

**Review by tsony-tsonev — APPROVED** [review]
LGTM

File: oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryUserRepository.java (hunk 1 above)
Comment (mgenov) [reply oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryUserRepository.java:31]: Good catch.

Comment (mgenov) [reply oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryUserRepository.java:31]: I'll add.

Comment (mgenov) [reply oauth2-example-app/src/main/java/com/clouway/oauth2/exampleapp/storage/InMemoryUserRepository.java:31]: Added.

**Pull request merged**